ocaml/xsh/xsh.ml

```diff
@@ -60,12 +60,24 @@ let proxy (ain : Unix.file_descr) (aout : Unix.file_descr) (bin : Unixfd.t)
         (if can_write a' then [bout] else [])
         @ if can_write b' then [aout] else []
       in
-      let r, w, _ = Unix.select r w [] (-1.0) in
-      (* Do the writing before the reading *)
-      List.iter
-        (fun fd -> if aout = fd then write_from b' a' else write_from a' b')
-        w ;
-      List.iter (fun fd -> if ain = fd then read_into a' else read_into b') r
+      let epoll = Polly.create () in
+      List.iter (fun fd -> Polly.add epoll fd Polly.Events.inp) r ;
+      List.iter (fun fd -> Polly.add epoll fd Polly.Events.out) w ;
+      Fun.protect
+        ~finally:(fun () -> Polly.close epoll)
+        (fun () ->
+          ignore
+          @@ Polly.wait epoll 4 (-1) (fun _ fd _ ->
```

Comment:
Note that the code below matches exactly one condition and not all fds that are ready; could this be written as
```
if fd = x then do something;
if fd = y then do something;
..
```

Maybe leave a comment why the selected behavior is the right one.